lightning/src/onion_message/dns_resolution.rs

```diff
@@ -211,16 +211,21 @@ pub struct HumanReadableName {
 impl HumanReadableName {
 	/// Constructs a new [`HumanReadableName`] from the `user` and `domain` parts. See the
 	/// struct-level documentation for more on the requirements on each.
-	pub fn new(user: &str, mut domain: &str) -> Result<HumanReadableName, ()> {
+	pub fn new(user: &str, domain: &str) -> Result<HumanReadableName, ()> {
 		// First normalize domain and remove the optional trailing `.`
-		if domain.ends_with('.') {
-			domain = &domain[..domain.len() - 1];
-		}
+		let domain = domain.strip_suffix(".").unwrap_or(domain);
```

Comment:
nit: `'.'` and `"."` are different - `'.'` is the `char` . whereas `"."` is the string ".". We should be stripping the char, not the string, as it uses a more efficient algorithm.